### PR TITLE
Export import

### PR DESCRIPTION
**Pull request by whitecat**
Adds import export functionality.

### CODE REVIEW

**Comment by whitecat** [comment]
There is a bug. In this setup we need to also reload the reading list. Right now we just reload the settings.

**Comment by whitecat** [comment]
Potential unintended consequence of import export. If we import and there are already existing reading links should we keep them or delete them? Currently I am deleting them.

File: js/options.js [hunk 1]
```diff
@@ -49,7 +49,68 @@ document.addEventListener('DOMContentLoaded', function () {
...
+        document.body.removeChild(link);
+      });
+    });
+  }
+
+  // Helper for import function
+  function impLoad() {
+    let myImportedData = JSON.parse(this.result);
+    //here is your imported data, and from here you should know what to do with it (save it to some storage, etc.)
+    chrome.storage.sync.clear();
```
Comment (alexpdraper) [inline js/options.js]: Getting rid of this line will preserve the existing data—but duplicates will still be overwritten (particularly the settings).

File: js/options.js [hunk 2]
```diff
@@ -49,7 +49,68 @@ document.addEventListener('DOMContentLoaded', function () {
...
+        link.dispatchEvent(event);
+        document.body.removeChild(link);
+      });
+    });
+  }
+
+  // Helper for import function
+  function impLoad() {
+    let myImportedData = JSON.parse(this.result);
+    //here is your imported data, and from here you should know what to do with it (save it to some storage, etc.)
```
Comment (alexpdraper) [inline js/options.js]: Copy/paste comment 😢

File: js/options.js [hunk 3]
```diff
@@ -49,7 +49,68 @@ document.addEventListener('DOMContentLoaded', function () {
...
+  function importFunc(e) {
+    let files = e.target.files
+    let reader = new FileReader();
+    reader.onload = impLoad;
+    reader.readAsText(files[0]);
+  }
+  
   restoreOptions();
+  // Import listeners
+  importOrig.addEventListener("change", importFunc, false);
```
Comment (alexpdraper) [inline js/options.js]: Double quotes! 😱

File: js/options.js [hunk 4]
```diff
@@ -49,7 +49,68 @@ document.addEventListener('DOMContentLoaded', function () {
     });
   }
 
+  // Helper for export function
+  let textFile = null;
```
Comment (alexpdraper) [inline js/options.js]: Using `let` is great, but it reduces the browser versions we support.

Chrome: 49+ (March 2, 2016)
Firefox: 44+ (Jan 26, 2016)

Is that something we want to do?

File: html/options.html [hunk 5]
```diff
@@ -25,5 +25,12 @@
...
 
+  <label data-localize="backup"></label>
+  <div>
+    <button id="exportBtn" data-localize="export"></button>
+    <button id="importBtn" data-localize="import"></button>
+    <input type="file" id="importOrig" accept=".json" style="display:none"/>
+  </div>
+
 </body>
 </html>
```
Comment (alexpdraper) [inline html/options.html]: Add a newline at the end of the file, please.

File: js/options.js (hunk 4 above)
Comment (whitecat) [reply js/options.js]: Maybe, We don't support before Firefox 53. 
But we do support all versions of Chrome. So I will remove it.

**Comment by whitecat** [comment]
Please review again!

**Pull request merged**